### PR TITLE
[StyleCop] Fix warnings on BotkitConversationState and BotkitDialogWrapper

### PR DESCRIPTION
**Pull request by Maihav**
Fix warnings on BotkitConversationState.cs and BotkitDialogWrapper.cs files.

**Changes**:
- Reorder constructors 
- Add class documentation 
- Remove blank lines
- Add documentation for methods and properties
- Remove unused async

### CODE REVIEW

File: Botkit/Microsoft.BotKit/BotkitDialogWrapper.cs [hunk 1]
```diff
@@ -1,51 +1,56 @@
-// Copyright(c) Microsoft Corporation.All rights reserved.
-// Licensed under the MIT License.
-
-using Microsoft.BotKit.Conversation;
-using Microsoft.Bot.Builder.Dialogs;
-using System;
+﻿// <copyright file="BotkitDialogWrapper.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
```
Comment (ceciliaavila) [inline Botkit/Microsoft.BotKit/BotkitDialogWrapper.cs]: Please, leave the copyright header as it was.

File: Botkit/Microsoft.BotKit/BotkitDialogWrapper.cs [hunk 2]
```diff
@@ -1,51 +1,56 @@
...
+        /// <summary>
+        /// Gets or sets an object containing variables and user responses from this conversation.
+        /// </summary>
+        public Tuple<string, object> Vars { get; set; }
+
         /// <summary>
         /// Jump immediately to the first message in a different thread.
         /// </summary>
-        /// <param name="thread">Name of a thread</param>
-        public async void GotoThread(string thread)
```
Comment (ceciliaavila) [inline Botkit/Microsoft.BotKit/BotkitDialogWrapper.cs]: I would leave these **_async_** indicators. The methods are not yet implemented, but, when they do, they should be async.

File: Botkit/Microsoft.BotKit/BotkitDialogWrapper.cs [hunk 3]
```diff
@@ -1,51 +1,56 @@
...
+        /// <param name="thread">Name of a thread.</param>
+        public void GotoThread(string thread)
         {
-
         }
 
         /// <summary>
         ///  Repeat the last message sent on the next turn.
         /// </summary>
-        public async void Repeat()
```
Comment (ceciliaavila) [inline Botkit/Microsoft.BotKit/BotkitDialogWrapper.cs]: Same here.

File: Botkit/Microsoft.BotKit/BotkitDialogWrapper.cs (hunk 1 above)
Comment (Maihav) [reply Botkit/Microsoft.BotKit/BotkitDialogWrapper.cs]: Done

File: Botkit/Microsoft.BotKit/BotkitDialogWrapper.cs (hunk 3 above)
Comment (Maihav) [reply Botkit/Microsoft.BotKit/BotkitDialogWrapper.cs]: ACK. Done

File: Botkit/Microsoft.BotKit/BotkitDialogWrapper.cs (hunk 2 above)
Comment (Maihav) [reply Botkit/Microsoft.BotKit/BotkitDialogWrapper.cs]: ACK. Done

File: Botkit/Microsoft.BotKit/BotkitConversationState.cs [hunk 4]
```diff
@@ -32,9 +42,9 @@ public string GetStorageKey(TurnContext context)
 
             // create a combo key by sorting all the fields in the conversation address and combining them all
             // mix in user id as well, because conversations are between the bot and a single user
-            const string ConversationId = "";//Object.keys(activity.conversation).sort().map((key) => activity.conversation[key]).filter((val) => val !== '' && val !== null && typeof val !== 'undefined').join('-') + '-' + activity.from.id;
+            const string ConversationId = ""; // Object.keys(activity.conversation).sort().map((key) => activity.conversation[key]).filter((val) => val !== '' && val !== null && typeof val !== 'undefined').join('-') + '-' + activity.from.id;
```
Comment (GasparAcevedoZainSouthworks) [inline Botkit/Microsoft.BotKit/BotkitConversationState.cs]: Remove this comment.

Comment (Maihav) [reply Botkit/Microsoft.BotKit/BotkitConversationState.cs]: ACK

Comment (Maihav) [reply Botkit/Microsoft.BotKit/BotkitConversationState.cs]: Done

File: libraries/Microsoft.BotKit/BotkitConversationState.cs [hunk 5]
```diff
@@ -1,12 +1,17 @@
-// Copyright(c) Microsoft Corporation.All rights reserved.
-// Licensed under the MIT License.
+﻿// <copyright file="BotkitConversationState.cs" company="PlaceholderCompany">
```
Comment (matiasroldan6) [inline libraries/Microsoft.BotKit/BotkitConversationState.cs]: Placeholder company??

File: libraries/Microsoft.BotKit/BotkitConversationState.cs [hunk 6]
```diff
@@ -15,15 +15,24 @@ namespace Microsoft.BotKit
     /// </summary>
     public class BotkitConversationState : ConversationState
     {
-        public BotkitConversationState(IStorage storage) : base(storage)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BotkitConversationState"/> class.
+        /// </summary>
+        /// <param name="storage">BotkitConversationState constructor.</param>
```
Comment (matiasroldan6) [inline libraries/Microsoft.BotKit/BotkitConversationState.cs]: This description doesn't belong to the `storage` paremeter

Comment (Maihav) [reply libraries/Microsoft.BotKit/BotkitConversationState.cs]: Done

**Pull request merged**